flink-runtime/src/test/java/org/apache/flink/runtime/externalresource/ExternalResourceUtilsTest.java

```diff
@@ -0,0 +1,247 @@
+		final Configuration config = new Configuration();
+		final String driverFactoryClassName = TestingExternalResourceDriverFactory.class.getName();
+		final PluginManager testingPluginManagerPluginManager = new TestingPluginManager(Collections.emptyMap());
+
+		config.set(ExternalResourceOptions.EXTERNAL_RESOURCE_LIST, Collections.singletonList(RESOURCE_NAME_1));
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX), driverFactoryClassName);
+
+		final Map<String, ExternalResourceDriver> externalResourceDrivers = ExternalResourceUtils.externalResourceDriversFromConfig(config, testingPluginManagerPluginManager);
+
+		assertThat(externalResourceDrivers.entrySet(), is(empty()));
+	}
+
+	@Test
+	public void testFactoryFailedToCreateDriver() {
+		final Configuration config = new Configuration();
+		final String driverFactoryClassName = TestingFailedExternalResourceDriverFactory.class.getName();
+		final Map<Class<?>, Iterator<?>> plugins = new HashMap<>();
+		plugins.put(ExternalResourceDriverFactory.class, IteratorUtils.singletonIterator(new TestingFailedExternalResourceDriverFactory()));
+		final PluginManager testingPluginManagerPluginManager = new TestingPluginManager(plugins);
+
+		config.set(ExternalResourceOptions.EXTERNAL_RESOURCE_LIST, Collections.singletonList(RESOURCE_NAME_1));
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX), driverFactoryClassName);
+
+		final Map<String, ExternalResourceDriver> externalResourceDrivers = ExternalResourceUtils.externalResourceDriversFromConfig(config, testingPluginManagerPluginManager);
+
+		assertThat(externalResourceDrivers.entrySet(), is(empty()));
+	}
+
+	@Test
+	public void testGetExternalResourceInfoProvider() {
+		final Configuration config = new Configuration();
+		final Map<Class<?>, Iterator<?>> plugins = new HashMap<>();
+		plugins.put(ExternalResourceDriverFactory.class, IteratorUtils.singletonIterator(new TestingExternalResourceDriverFactory()));
+		final PluginManager testingPluginManagerPluginManager = new TestingPluginManager(plugins);
+
+		final String driverFactoryClassName = TestingExternalResourceDriverFactory.class.getName();
+		config.set(ExternalResourceOptions.EXTERNAL_RESOURCE_LIST, Collections.singletonList(RESOURCE_NAME_1));
+		config.setLong(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_AMOUNT_SUFFIX), RESOURCE_AMOUNT_1);
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX), driverFactoryClassName);
+
+		final ExternalResourceInfoProvider externalResourceInfoProvider = ExternalResourceUtils.getExternalResourceInfoProvider(config, testingPluginManagerPluginManager);
```

Comment:
`getExternalResourceInfoProvider` would be a bit easier to test if the method did a little bit less. We could, for example, pass in the `externalResourceAmountMap` and `externalResourceDrivers`. We could also introduce an internal method which does the job. That way we don't have to set up all the `PluginManager` things. Moreover, the loading of `Drivers` and the creation of the `externalResourceAmountMap` has already been tested by other methods.